activestorage/lib/active_storage/service/mirror_service.rb

```diff
@@ -30,6 +30,13 @@ def self.build(primary:, mirrors:, name:, configurator:, **options) # :nodoc:
 
     def initialize(primary:, mirrors:)
       @primary, @mirrors = primary, mirrors
+      @executor = Concurrent::ThreadPoolExecutor.new(
```

Comment:
Can you do a performance benchmarking?

Reply:
I'll add performance benchmark and request review to you.

Reply:
@akhilgkrishnan 
This is performance benchmark using GCS. In my opinion, there is a performance improvement when looking at real time.

- repeat time: 10
- Used service: Google Cloud Storage (region: ap-northeast-3)
- Used image size: 512kb

- primary: 1, mirrors: 2

| Task       | Real Time | User Time | System Time | Total Time |
|------------|-----------|-----------|-------------|------------|
| Parallel   | 0.453     | 0.055     | 0.015       | 0.070      |
| Sequential | 0.979     | 0.045     | 0.011       | 0.056      |

- primary: 1, mirrors: 1

| Task       | Real Time | User Time | System Time | Total Time |
|------------|-----------|-----------|-------------|------------|
| Parallel   | 0.380     | 0.053     | 0.020       | 0.074      |
| Sequential | 0.624     | 0.033     | 0.008       | 0.041      |

Reply:
@akhilgkrishnan Gentle ping 😄 Could you please review when you get a chance?

Reply:
@akhilgkrishnan Gentle ping 🙏

Reply:
Sorry @heka1024 for the late response. This looks good to me. Lets wait someone from core to review.